thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/views/function/SplineRegressionAnomalyTimeSeriesView.java

```diff
@@ -0,0 +1,166 @@
+    // The baseline for SplineRegression is 2-3 months of continuous data.
+    int historicalDataMaxLength = Integer.parseInt(
+        properties.getProperty(HISTORICAL_DATA_LENGTH, DEFAULT_HISTORICAL_DATA_LENGTH));
+    TimeUnit bucketUnit =
+        TimeUnit.valueOf(properties.getProperty(BUCKET_UNIT, DEFAULT_BUCKET_UNIT));
+    long trainingStartTimeMillis = 0L;
+    // Daylight saving time only affects data ranges when TimeUnit is larger than HOURS. Since
+    // there is only one unit, i.e., DAYS, that is larger than HOURS, it is sufficient to check
+    // if user's unit equals to DAYS.
+    if (TimeUnit.DAYS.equals(bucketUnit)) {
+      String timeZoneString = properties.getProperty(TIMEZONE, DEFAULT_TIMEZONE);
+      DateTimeZone timeZone = DateTimeZone.forID(timeZoneString);
+      DateTime trainingStartTime = new DateTime(monitoringWindowStartTime, timeZone);
+      Period period = new Period(0, 0, 0, historicalDataMaxLength, 0, 0, 0, 0);
+      trainingStartTime = trainingStartTime.minus(period);
+      trainingStartTimeMillis = trainingStartTime.getMillis();
+    } else {
+      long startOffset = bucketUnit.toMillis(historicalDataMaxLength);
+      trainingStartTimeMillis = monitoringWindowStartTime - startOffset;
+    }
+    startEndTimeIntervals.add(new Pair<>(trainingStartTimeMillis, monitoringWindowStartTime));
+
+    return startEndTimeIntervals;
+  }
+
+  /**
+   * The baseline of the returned time series is the average of the values in past weeks, which is defined by
+   * baselineSeasonalPeriod.
+   *
+   */
+  @Override
+  public AnomalyTimelinesView getTimeSeriesView(MetricTimeSeries timeSeries, long bucketMillis, String metric,
+      long viewWindowStartTime, long viewWindowEndTime, List<RawAnomalyResultDTO> knownAnomalies) {
+
+    AnomalyTimelinesView anomalyTimelinesView = new AnomalyTimelinesView();
+
+    SplineRegressionAnomalyDetectionFunction splineRegressionFunction =
+        new SplineRegressionAnomalyDetectionFunction();
+    try {
+      splineRegressionFunction.init(properties);
+    }
+    catch (IllegalFunctionException e){
+      LOG.error("{}: Unable to receive configure file", e.getMessage());
+    }
+
+    TimeRange monitorTimeRange = new TimeRange(viewWindowStartTime, viewWindowEndTime);
+
+    List<AnomalyResult> anomalyHistory = new ArrayList<>();
+    if(knownAnomalies != null && !knownAnomalies.isEmpty()) {
+      for (RawAnomalyResultDTO anomaly : knownAnomalies) {
+        AnomalyFeedbackDTO feedbackDTO = anomaly.getFeedback(); // null if no user feedback
+        TimeRange anomalyTimeRange = new TimeRange(anomaly.getStartTime(), anomaly.getEndTime());
+        AnomalyResult result = new AnomalyResult((feedbackDTO == null), anomalyTimeRange, anomaly.getScore(), Math.abs(anomaly.getWeight()), AnomalyResult.AnomalyMethod.SPLINEREGRESSION, properties);
+        anomalyHistory.add(result);
+      }
+    }
+
+    MapMetricTimeSeries series = new MapMetricTimeSeries();
```

Comment:
I am not sure if it is a good idea to directly access anomaly detection library here. We need a property interface to communicate this internal libraries.